litho-processor/src/main/java/com/facebook/litho/specmodels/generator/EventGenerator.java

```diff
@@ -119,16 +122,81 @@ static TypeSpecDataHolder generateEventDispatcher(EventDeclarationModel eventDec
             .addModifiers(Modifier.STATIC)
             .addParameter(ClassNames.EVENT_HANDLER, "_eventHandler");
 
-    eventDispatcherMethod.addStatement(
-        "final $T _eventState = new $T()", eventDeclaration.name, eventDeclaration.name);
+    final TypeElement eventRepresentedObject = (TypeElement) eventDeclaration.representedObject;
+    final TypeElementCtorsInfo eventClassCtorsInfo =
+        TypeElementUtils.extractCtorsInfo(eventRepresentedObject);
+
+    /*
+    * Check if our class has any non-empty constructors. If so we will instantiate the
+    * Event class using that one instead of accessing mutable fields.
+    * */
+    if (eventClassCtorsInfo == TypeElementCtorsInfo.SINGLE_NON_EMPTY) {
+      final StringBuilder ctorParamsPlaceholder = new StringBuilder();
+      final List<String> ctorParamNames = new ArrayList<>();
+
+      /*
+      * Iterate over the Event fields in order to achieve the following:
+      *
+      * 1) Add the parameters to the method under construction.
+      * 2) Create a parameter name placeholder for JavaPoet.
+      * 3) Keep a list of all parameter names.
+      * */
+      for (FieldModel fieldModel : eventDeclaration.fields) {
+        if (fieldModel.field.modifiers.contains(Modifier.STATIC)
+            && fieldModel.field.modifiers.contains(Modifier.FINAL)) {
+          continue;
+        }
+
+        eventDispatcherMethod
+            .addParameter(fieldModel.field.type, fieldModel.field.name);
+
+        // We do not want ',' appended on the first loop
+        if (!ctorParamNames.isEmpty()) {
+          ctorParamsPlaceholder.append(", ");
+        }
+
+        ctorParamsPlaceholder.append("$L");
 
-    for (FieldModel fieldModel : eventDeclaration.fields) {
-      if (fieldModel.field.modifiers.contains(Modifier.FINAL)) {
-        continue;
+        ctorParamNames.add(fieldModel.field.name);
       }
+
+      /*
+      * Create a list with all the desired arguments to be for the following statement.
+      * These need to be the following:
```

Comment:
Let's maybe reword a bit?
"
Create a list of all arguments for Event instantiation statement below. Should be the following:
1) Event class name (type declaration)
2) Event class name (constructor call)
3) Event constructor parameters' names
"